src/toil/common.py

```diff
@@ -524,3 +530,36 @@ def parseSetEnv(l):
             raise ValueError('Empty name')
         d[k] = v
     return d
+
+
+def getToilWorkflowDir(workflowID, workDir=None):
+    """
+    Returns a path to the directory where worker directories and the cache will be located for this
+    workflow.  This is particularly useful if the user hasn't provided a value for --workDir (passed
+    to this function as workDir) and the TOIL_WORKDIR environment variable is unset.  In this case,
+    the value returned points to a directory 'toil' located somewhere on the system in a location
+    reserved for temporary files (obtained from the tempfile module).  The variable is created if
+    required.
+    Hierarchy for workdir is config.workDir >> $TOIL_WORKDIR >> $TMPDIR
+    :param workflowID: Unique identifier for the workflow
+    :param workDir: Value passed to the program using the workDir flag
+    :return: Path to the work dir
+    """
+    workDir = workDir or os.getenv('TOIL_WORKDIR')
+    if workDir is None:
```

Comment:
Looks like this doesn't do what we agreed on: https://github.com/BD2KGenomics/toil/pull/692#issuecomment-193533704

We want the workflow dir to always be created, right?